quic/s2n-quic-core/src/lib.rs

```diff
@@ -30,6 +30,16 @@ macro_rules! assume {
     };
 }
 
+/// Checks that the first argument is true, otherwise returns the second value
+#[macro_export]
+macro_rules! ensure {
```

Comment:
Do you think this naming makes sense for someone not familiar with the s2n-tls version? What about something like `assert_or`?

Reply:
So for s2n-tls I actually stole ENSURE and BAIL from macros in the rust ecosystem 😅

https://docs.rs/anyhow/latest/anyhow/macro.ensure.html
https://docs.rs/anyhow/latest/anyhow/macro.bail.html

https://docs.rs/failure/latest/failure/macro.ensure.html
https://docs.rs/failure/latest/failure/macro.bail.html